### PR TITLE
Escape key now brings up a dialog to confirm you want to stop emulation

### PR DESCRIPTION
**Pull request by EliseZeroTwo**
Solves #1042

### CODE REVIEW

File: Ryujinx/Ui/GLRenderer.cs [hunk 1]
```diff
@@ -138,7 +138,8 @@ public void HandleScreenState(KeyboardState keyboard)
                     {
                         if (keyboard.IsKeyDown(OpenTK.Input.Key.Escape))
                         {
-                            Exit();
+                            if (GtkDialog.CreateExitDialog())
+                                Exit();
```
Comment (riperiperi) [inline Ryujinx/Ui/GLRenderer.cs]: Prefer if with brackets to one-line.

File: Ryujinx/Ui/GtkDialog.cs [hunk 2]
```diff
@@ -29,5 +29,33 @@ internal static void CreateErrorDialog(string errorMessage)
         {
             CreateDialog("Ryujinx - Error", "Ryujinx has encountered an error", errorMessage);
         }
+
+        internal static bool isExitDialogOpen = false;
+        internal static bool CreateExitDialog()
+        {
+            if (isExitDialogOpen)
```
Comment (riperiperi) [inline Ryujinx/Ui/GtkDialog.cs]: Prefer if with brackets to single line if.

File: Ryujinx/Ui/GtkDialog.cs [hunk 3]
```diff
@@ -29,5 +29,33 @@ internal static void CreateErrorDialog(string errorMessage)
...
+            MessageDialog messageDialog = new MessageDialog(null, DialogFlags.Modal, MessageType.Question, ButtonsType.OkCancel, null)
+            {
+                Title = "Ryujinx - Exit",
+                Icon = new Gdk.Pixbuf(Assembly.GetExecutingAssembly(), "Ryujinx.Ui.assets.Icon.png"),
+                Text = "Are you sure you want to stop emulation?",
+                SecondaryText = "All unsaved data will be lost",
+                WindowPosition = WindowPosition.Center
+            };
+            messageDialog.SetSizeRequest(100, 20);
+            ResponseType res = (ResponseType)messageDialog.Run();
```
Comment (riperiperi) [inline Ryujinx/Ui/GtkDialog.cs]: This will block the calling thread, right? Any adverse affect on the GLRenderer/Game/Input?

Comment (EliseZeroTwo) [reply Ryujinx/Ui/GtkDialog.cs]: This doesn't seem to affect anything beyond input to the main window (while the dialog is open), I was able to load and run Celeste fine with the dialog open. That's the best I can answer this, sorry!

File: Ryujinx/Ui/GtkDialog.cs [hunk 4]
```diff
@@ -29,5 +29,35 @@ internal static void CreateErrorDialog(string errorMessage)
...
         }
+
+        internal static bool isExitDialogOpen = false;
+        internal static bool CreateExitDialog()
+        {
+            if (isExitDialogOpen)
+            {
+                return false;
+            }
+            isExitDialogOpen = true;
```
Comment (AcK77) [inline Ryujinx/Ui/GtkDialog.cs]: Add a new empty line before

File: Ryujinx/Ui/GtkDialog.cs [hunk 5]
```diff
@@ -29,5 +29,35 @@ internal static void CreateErrorDialog(string errorMessage)
...
+                Title = "Ryujinx - Exit",
+                Icon = new Gdk.Pixbuf(Assembly.GetExecutingAssembly(), "Ryujinx.Ui.assets.Icon.png"),
+                Text = "Are you sure you want to stop emulation?",
+                SecondaryText = "All unsaved data will be lost",
+                WindowPosition = WindowPosition.Center
+            };
+            messageDialog.SetSizeRequest(100, 20);
+            ResponseType res = (ResponseType)messageDialog.Run();
+            messageDialog.Dispose();
+            isExitDialogOpen = false;
```
Comment (AcK77) [inline Ryujinx/Ui/GtkDialog.cs]: Add a new empty line before

**Comment by EliseZeroTwo** [comment]
@AcK77 fixed, thanks

File: Ryujinx/Ui/GtkDialog.cs [hunk 6]
```diff
@@ -29,5 +29,37 @@ internal static void CreateErrorDialog(string errorMessage)
         {
             CreateDialog("Ryujinx - Error", "Ryujinx has encountered an error", errorMessage);
         }
+
+        internal static bool isExitDialogOpen = false;
```
Comment (Thog) [inline Ryujinx/Ui/GtkDialog.cs]: nit: Add one empty line after this

Comment (jduncanator) [reply Ryujinx/Ui/GtkDialog.cs]: This variable should also be capitalised, as it's a member, not a local.

Comment (EliseZeroTwo) [reply Ryujinx/Ui/GtkDialog.cs]: Fixed

**Pull request merged**